lib/vanagon/platform/dsl.rb

```diff
@@ -221,13 +221,70 @@ def docker_image(name)
         @platform.docker_image = name
       end
 
+      # Set the ami for the platform to use
+      #
+      # @param ami [String] the ami id used.
+      def aws_ami(ami)
+        @platform.aws_ami = ami
+      end
+
+      # Set the user data used in AWS to do setup. Like cloud-config
+      #
+      # @param userdata [String] a string used to send to the node to do the intial setup
+      def aws_user_data(userdata)
+        @platform.aws_user_data = userdata
+      end
+
+      # Set the region, this defaults to us-east-1
+      #
+      # @param region [String] a string used to setup the region
+      def aws_region(region = 'us-east-1')
+        @platform.aws_region = region
+      end
+
+      # Set the shutdown behavior for aws. This will default to terminate the instance on shutdown
+      #
+      # @param shutdown_behavior [String] a string used to set the shutdown behavior
+      def aws_shutdown_behavior(shutdown_behavior = 'terminate')
```

Comment:
Since there's a default value, I assume there's also a list of acceptable values? Can we check and reject any values that might be unacceptable here?